internals/review_models.py

```diff
@@ -188,3 +188,131 @@ def get_raw_owner_file(cls, url):
       return None
 
     return owners_file.raw_content
+
+
+# Note: This class is not used yet.
+class Gate(ndb.Model):  # copy from ApprovalConfig
+  """Gates regulate the completion of a stage."""
+  feature_id = ndb.IntegerProperty(required=True)
+  stage_id = ndb.IntegerProperty(required=True)
+  gate_type = ndb.IntegerProperty(required=True)  # copy from field_id
+
+  # Can be REVIEW_REQUESTED or one of ApprovalValue states
+  state = ndb.IntegerProperty(required=True)  # calc from Approval
+
+  owners = ndb.StringProperty(repeated=True)
+  next_action = ndb.DateProperty()
+  additional_review = ndb.BooleanProperty(default=False)
+
+  # TODO(jrobbins): implement request_review() and clear_request()
+
+
+# Note: This class is not used yet.
+class Vote(ndb.Model):  # copy from Approval
+  """One approver's vote on what the state of a gate should be."""
+
+  # Not used: NEEDS_REVIEW = 0
+  NA = 1
+  REVIEW_REQUESTED = 2
+  REVIEW_STARTED = 3
+  NEED_INFO = 4
+  APPROVED = 5
+  NOT_APPROVED = 6
+  VOTE_VALUES = {
+      # Not used: NEEDS_REVIEW: 'needs_review',
+      NA: 'na',
+      REVIEW_REQUESTED: 'review_requested',
+      REVIEW_STARTED: 'review_started',
+      NEED_INFO: 'need_info',
```

Comment:
If DevRel and product are going to approve summaries, we need a vote value called `NEED_REVISION`.

Or maybe a value called `NEED_CHANGE` could apply to both.

Reply:
Since our users will familiar with Monorail FLT and will also be using go/launch at the same time, we need to follow the conventions used in those tools.  Basically `NEED_INFO` means that something is needed from the feature owner, it could be info or a revision.

Reply:
I looked at go/launch again and they use the term "Needs Work", so we will go with that.  I think that is consistent with a reviewer asking for changes or information.  Monorail FLT uses "Needs info", however going forward our users will be concurrently using go/launch, so aligning with that tool is more important.  I am sending a separate small PR.

Reply:
👍